### ISSUE TITLE
iOS 16 and XCode 14 crash because of kCFStreamNetworkServiceTypeVoIP

### ISSUE THREAD

**Issue by AmarOk1412** [pjsip/pjproject#3303]
### Describe the bug

Cf context for more details.

### Steps to reproduce

1. Build pjsip for iOS 16 (XCode 14)
2. Run the app
=> Crash

> Application Specific Information: Linked against modern SDK, VOIP socket will not wake. Use Local Push Connectivity instead

### PJSIP version

2.12

### Context

iOS 16

In pjlib/src/pj/activesock.c

```c
CFReadStreamSetProperty(asock->readStream,
                                    kCFStreamNetworkServiceType,
                                    kCFStreamNetworkServiceTypeVoIP)
```

This is not supported anymore

### Log, call stack, etc

Other sources:

+ https://stackoverflow.com/questions/73703471/voip-socket-on-ios-16-and-xcode-14
+ https://github.com/bluejeans/ios-client-sdk/issues/12
+ https://gitlab.linphone.org/BC/public/belle-sip/-/blob/6df7534aa3213f3a27a054e7f76a79778c5b3265/src/transports/stream_channel.c#L225

Temporary workaround: Disabling PJ_IPHONE_OS_HAS_MULTITASKING_SUPPORT

**Comment by AmarOk1412** [comment]
hmm will try `Set PJ_ACTIVESOCK_TCP_IPHONE_OS_BG to 0, or alternatively call pj_activesock_enable_iphone_os_bg(PJ_FALSE) before creating any SIP transport or any PJSIP socket in general, this is for disabling the VOIP socket. `

**Comment by AmarOk1412** [comment]
Why not just add

```
 pjlib/include/pj/compat/os_auto.h.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/pjlib/include/pj/compat/os_auto.h.in b/pjlib/include/pj/compat/os_auto.h.in
index 10e08f0bd..dd5e642ea 100644
--- a/pjlib/include/pj/compat/os_auto.h.in
+++ b/pjlib/include/pj/compat/os_auto.h.in
@@ -199,7 +199,11 @@
  	    /* Is multitasking support available?  (see ticket #1107) */
 #	    define PJ_IPHONE_OS_HAS_MULTITASKING_SUPPORT 	1
 	    /* Enable activesock TCP background mode support */
-#	    define PJ_ACTIVESOCK_TCP_IPHONE_OS_BG		1
+#         ifdef __IPHONE_16_0
+#	        define PJ_ACTIVESOCK_TCP_IPHONE_OS_BG		0
+#         else
+#	        define PJ_ACTIVESOCK_TCP_IPHONE_OS_BG		1
+#	endif
 #	endif
 #    endif
 #endif
```

In the config? As anyway it will crash without?

**Comment by sauwming** [comment]
Yes, we already did something like that in https://github.com/pjsip/pjproject/pull/3253, which is included in the newly released 2.13.

**Comment by rjvishnu87** [comment]
@sauwming If I add this crash fix in PJSIP 2.10, should It work or not ? Because I am still using 2.10 version.

**Comment by sauwming** [comment]
Haven't tried it, but yes, I believe it should.
